### ISSUE TITLE
[cpp] "identifier" should be `\grammarterm{identifier}` in more cases

### ISSUE THREAD

**Issue by Eisenwave** [cplusplus/draft#6524]
The [[cpp]](https://eel.is/c++draft/cpp) section talks about "identifier"s frequently; often in a context where direct statements about grammar and macro expansion are made.

It would be appropriate to use `\grammarterm{identifier}` in many more places. For example:

> A preprocessing directive of the form
> 
> `#` `undef` *identifier* *new-line*
> 
> causes the specified identifier no longer to be defined as a macro name.
> It is ignored if the specified identifier is not currently defined as a macro name.

\- [[cpp.scope] p2](https://eel.is/c++draft/cpp.scope#2)

This is one of the more extreme cases: statements directly about grammar rules are made, but "identifier" is not formatted as a grammatical term.

**Comment by jensmaurer** [comment]
Yes, that's a heritage from C where grammar non-terminals are/were directly used as prose terms. I'm not surprised the preprocessor section is particularly visible offender.

**Comment by AlisdairM** [comment]
It might also be helpful to introduce a separate _pp-identifier_ token, to distinguish between when an identifier is a _pp-token_ vs when it is a _token_.
